collect_app/src/test/java/org/odk/collect/android/projects/ProjectsRepositoryTest.kt

```diff
@@ -0,0 +1,67 @@
+package org.odk.collect.android.projects
+
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.`is`
+import org.hamcrest.Matchers.not
+import org.hamcrest.Matchers.notNullValue
+import org.junit.Before
+import org.junit.Test
+
+abstract class ProjectsRepositoryTest {
+    lateinit var projectsRepository: ProjectsRepository
+
+    abstract fun buildSubject(): ProjectsRepository
+
+    @Before
+    fun setup() {
+        projectsRepository = buildSubject()
+    }
+
+    @Test
+    fun getAll_shouldReturnAllProjectsFromStorage() {
+        projectsRepository.add("ProjectX")
+        projectsRepository.add("ProjectY")
+        projectsRepository.add("ProjectZ")
+
+        val projects = projectsRepository.getAll()
+
+        assertThat(projects.size, `is`(3))
+        assertThat(projects[0].name, `is`("ProjectX"))
```

Comment:
We're implicitly testing order here. Not a problem right now but in the future we might just want to use the `contains` matcher if we don't care about ordering on the `getAll` results.